share/smack/include/smack-macros.h

```diff
@@ -0,0 +1,142 @@
+// shamelessly stole these macro definitions from https://github.com/pfultz2/Cloak/wiki/C-Preprocessor-tricks,-tips,-and-idioms
```

Comment:
Please add LICENSE info to this new file. Also, break this first long comment into two lines.